libraries/Microsoft.Bot.Builder.LanguageGeneration/Extensions.cs

```diff
@@ -147,6 +153,51 @@ public static string GetPrefixErrorMessage(this LGTemplateParser.NormalTemplateS
             return errorPrefix;
         }
 
+        /// <summary>
+        /// Generator random seed and value from properties.
+        /// If value is not null, the mock random value result would be: min + (value % (max - min)).
+        /// Else if seed is not null, the seed of the random would be fixed.
+        /// </summary>
+        /// <param name="properties">Properties.</param>
+        /// <param name="min">The inclusive lower bound of the random number returned.</param>
+        /// <param name="max">The exclusive upper bound of the random number returned. max must be greater than or equal to min.</param>
+        /// <param name="seed">user seed.</param>
+        /// <returns>Random seed and value.</returns>
+        public static int GeneratorMockRandom(this IDictionary<string, object> properties, int min, int max, int? seed = null)
+        {
```

Comment:
This should be in Dialog.Adaptive and not in Language Generation.  The reason is dependencies--we don't want selectors depending on LG.